theano/scan_module/scan_perform.pyx

```diff
@@ -514,7 +514,8 @@ def perform(
             if i == 0:
                 jout = j+offset_out
                 shape = (store_steps[j],) + output_storage[jout].storage[0].shape
-                if len(output_storage[jout].storage[0].shape) == 0:
+                if output_storage[jout].storage[0].ndim == 0 and \
+                    isinstance(output_storage[jout].storage[0], numpy.ndarray):
```

Comment:
I discussed with Pierre Luc and we think you can just delete the 2 lines in the original file 517 and 518. Can you do that in this file and in the file scan_op.py for the non cython version? If all tests pass, then it is good.

Those 2 lines dates from 2011 and since that, we re-worked how vector_outs work and those 2 lines aren't needed anymore.

Having those 2 lines don't cause wrong results, that is why it wasn't found before. You did a good finding! thanks.